### PR TITLE
Projecttransform rebase

### PR DESCRIPTION
**Pull request by Pomax**
supercedes https://github.com/mozilla/webmaker-android/pull/2264

### CODE REVIEW

**Comment by Pomax** [comment]
@thisandagain, @xmatthewx can I ask you two to see what you think about the feel of this project view zoom. It's not a strict "to/from center of pinch zooming", but does quite a good job (I think) of letting you find your content by zooming the content "towards and through" the pinch center instead.

**Comment by xmatthewx** [comment]
This PR seems too big and risky to land at the last minute. And, I haven't had a chance to test on device. We should put this on hold unless someone else has an abundance of time and confidence. cc @gvn @k88hudson

**Comment by Pomax** [comment]
emphasis on "seems", the refactor is for the most part a true refactor: the code is exactly the same, just moved to their own files to keep development sane. The only real rewrites are in transform.js

File: www_src/pages/project/transforms.js [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    startDY = 0;
+    startDistance = undefined;
+    currentX = undefined;
+    currentY = undefined;
+    currentZoom = undefined;
+  };
+
+  // Swipe handling is supposed to kick in for... something...
+  var handleSwipe = () => {
+    // FIXME: TODO: turned off for now, in part because of Laura's comments, in
```
Comment (Pomax) [inline www_src/pages/project/transforms.js]: not sure if we want to keep this disabled or reenable it. On a nexus 7, the swiping does pretty much nothing visually different from regular translation

**Comment by xmatthewx** [comment]
@Pomax - good to know. We also "seem" to be short on hands and eyes today, but we can keep the door open and try to land this.

**Comment by xmatthewx** [comment]
That last comment sounds snarky. Not intended. Personally, I will be thrilled if we can include this ... it will greatly improve demos in Whistler.

**Comment by k88hudson** [comment]
@Pomax Really excellent work here. However since this is mostly a refactor/lays the groundwork for future improvements rather than directly impacting the user experience, I would vote to land post-launch or on a separate branch. Does that sound OK to you?

**Comment by gvn** [comment]
I also vote for landing post-launch.

**Comment by thisandagain** [comment]
Thanks for putting this together @Pomax 

I agree with landing this after the Whistler release tomorrow as it introduces quite a bit of risk. Including it in the nightly should give us lots of time to test on-device.

File: www_src/pages/project/transforms.js (hunk 1 above)
Comment (thisandagain) [inline www_src/pages/project/transforms.js]: This works in `play` mode currently. Disabling it feels like a regression. Laura's notes were specific to the UK ... users in Bangladesh understood it.

**Comment by thisandagain** [comment]
Just did one more review / walk-through. This is R+ as long as we can re-enable swiping between views in play mode.

**Comment by thisandagain** [comment]
R+ Thanks a ton @Pomax

**Pull request merged**